client/ayon_core/pipeline/context_tools.py

```diff
@@ -579,3 +586,39 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def version_up_current_workfile():
+    """Function to increment and save workfile
+    """
+    host = registered_host()
+    project_name = get_current_project_name()
+    folder_path = get_current_folder_path()
+    task_name = get_current_task_name()
+    host_name = get_current_host_name()
+
+    template_key = get_workfile_template_key_from_context(
+        project_name,
+        folder_path,
+        task_name,
+        host_name,
+    )
+    anatomy = Anatomy(project_name)
+
+    data = get_template_data_with_names(
+        project_name, folder_path, task_name, host_name
+    )
+    data["root"] = anatomy.roots
+
+    work_template = anatomy.get_template_item("work", template_key)
+
+    # Define saving file extension
+    extensions = host.get_workfile_extensions()
+
+    work_root = work_template["directory"].format_strict(data)
+    file_template = work_template["file"].template
+    last_workfile_path = get_last_workfile(
+        work_root, file_template, data, extensions, True
+    )
```

Comment:
This seems to, in no way, rely on any current filename even if there were. We may need to somehow preserve the current file extension if the host DOES have a filename, so that if e.g. in maya the current scene is `.ma` it'll still be saved as `.ma` and if it is `.mb` the next save is also `.mb`.

Aside of that of course it's still a question that if the file is unsaved - what should it default to then? I suppose the same as what workfiles tool would?

Reply:
I have added the conditions to check to see if there is current file. If there is one, it will follow the extension of the current workfile. If there isn't, it will use .ma extension.